boards/arm/nrf52840_pca10059/Kconfig.defconfig

```diff
@@ -24,6 +24,17 @@ config FLASH_LOAD_OFFSET
 	default 0x1000
 	depends on BOARD_HAS_NRF5_BOOTLOADER && !USE_DT_CODE_PARTITION
 
+if SERIAL
+
+choice UART_0_NRF_TYPE
```

Comment:
This is currently not working actually because `UART_0_NRF_TYPE` is set as `optional`. @ulfalizer is there something we can do about this?

Reply:
What's the intention? That the choice should become enabled and default to `UART_0_NRF_UARTE` if `SERIAL` is enabled?

To do that, `UART_0_NRF_UARTE` would need to be enabled in a configuration file. You'll get a warning if `SERIAL` isn't enabled at the same time though.